src/main/java/com/wojteknier03/clinic_medical/mapper/DoctorMapper.java

```diff
@@ -0,0 +1,34 @@
+package com.wojteknier03.clinic_medical.mapper;
+
+import com.wojteknier03.clinic_medical.dto.DoctorDto;
+import com.wojteknier03.clinic_medical.model.Clinic;
+import com.wojteknier03.clinic_medical.model.Doctor;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Named;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Mapper(componentModel = "spring")
+public interface DoctorMapper {
+    @Mapping(target = "clinicIds", source = "clinics", qualifiedByName = "clinicsToClinicIds")
+    DoctorDto toDto(Doctor doctor);
+
+    @Mapping(target = "clinics", ignore = true)
+    Doctor fromDto(DoctorDto doctorDto);
+
+    @Named("clinicsToClinicIds")
+    default Set<Long> clinicsToClinicIds (Set<Clinic> clinics) {
+        return clinics.stream()
+                .map(Clinic::getId)
+                .collect(Collectors.toSet());
+    }
+
+    default List<DoctorDto> toDtoList(List<Doctor> doctors){
```

Comment:
ta metoda nie jest nigdzie uzywana? nie ma nad soba adnotacji named ani qualifiedByName nigdzie w metodach mapujacych rowniez nie masz. Po co ona jest?

Reply:
nie piszemy implementacji jesli chcesz metode mapujaca liste po prostu zdefiniuj jej sygnature i tyle. Nie pisz implementacji to mapstruct ma nam implementacje wygenerowac